### PR TITLE
uwsgi: fix freeze, http/1.1 and keepalive

### PR DESCRIPTION
**Pull request by sileht**
uwsgi: fix freeze, http/1.1 and keepalive
uwsgi: --thunder-lock doesn't take param in cmd

### CODE REVIEW

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 5fa2cd41e98b364c5ae1462a257f9033e266924d

File: doc/source/influxdb.rst [hunk 1]
```diff
@@ -39,5 +39,14 @@ configuration example::
     http_headers = {"X-Gnocchi-InfluxDB-Tag-Resource-ID" = "host"}
 
 
+Gnocchi configuration
+=====================
+
+The default Gnocchi-API web server does not support the chunked encoding
```
Comment (jd) [inline doc/source/influxdb.rst]: Gnocchi API server

File: doc/source/influxdb.rst [hunk 2]
```diff
@@ -39,5 +39,14 @@ configuration example::
...
+required by influxdb. To enable chunked encoding, you must put a real HTTP
```
Comment (jd) [inline doc/source/influxdb.rst]: the InfluxDB compatible endpoint.

File: doc/source/influxdb.rst [hunk 3]
```diff
@@ -39,5 +39,14 @@ configuration example::
...
+Server (apache/nginx/...) on front of Gnocchi-API, and set
```
Comment (jd) [inline doc/source/influxdb.rst]: Apache/NGINX/…
Gnocchi API

File: gnocchi/opts.py [hunk 4]
```diff
@@ -64,6 +64,15 @@ def __getitem__(self, key):
     cfg.PortOpt('port',
                 default=8041,
                 help="Port to listen on"),
+    cfg.StrOpt('uwsgi-mode',
+               default='http',
+               choices=["http", "http-socket", "socket"],
+               help="""Socket type to use for uwsgi:
```
Comment (jd) [inline gnocchi/opts.py]: uWSGI

File: gnocchi/opts.py [hunk 5]
```diff
@@ -64,6 +64,15 @@ def __getitem__(self, key):
...
+* http: support HTTP/1.1 and keepalive,
+  but not chunked encoding (eg: influxdb)
```
Comment (jd) [inline gnocchi/opts.py]: (InfluxDB)

It's the only thing using it so no need for e.g.

File: gnocchi/opts.py [hunk 6]
```diff
@@ -64,6 +64,15 @@ def __getitem__(self, key):
...
+* http-socket/socket: support chunked encoding, but require a upstream HTTP
```
Comment (jd) [inline gnocchi/opts.py]: That does not tell what's the difference between http-socket and socket thought

File: gnocchi/rest/app.py [hunk 7]
```diff
@@ -62,6 +62,15 @@ def on_route(self, state):
         state.request.policy_enforcer = self.policy_enforcer
         state.request.auth_helper = self.auth_helper
 
+    def after(self, state):
```
Comment (jd) [inline gnocchi/rest/app.py]: @staticmethod

File: gnocchi/opts.py (hunk 6 above)
Comment (sileht) [reply gnocchi/opts.py]: They are identical for us, while it depends on how you configure apache/nginx:
* http-socket: nginx<->uwsgi talks http protocol
* socket: nginx<->uwsgi talks uwsgi protocol

That's why I didn't put more detail.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 8e844aaa403fb073326bdb71b3e9f3d41161fcb1

**Pull request merged**